client-side-virtual-keyboard.md

```diff
@@ -24,87 +23,46 @@ After importing the AAR file, use the AnboxWebView in an Android project:
-
-   ```
-       window.stream = new AnboxStream({
-           ...
-           ...
-           ...
-           callbacks: {
-               ...
-               ...
-               messageReceived: (type, data) => {
-                   switch(type) {
-                   case "show-ime" :
-                       AppInterface.openVirtualKeyboard();
-                   break;
-                   case "hide-ime" :
-                       AppInterface.hideVirtualKeyboard();
-                   break;
-                   default:
-                       console.log("Unhandled message:" + type)
-                   break;
-                   }
-               }
-           }
-       });
-   ```
-
-4. Set up the listener to the activity where you want to capture the message. You must implement the following methods for the AppInterface.ActionListener interfaces so that the application will be notified when a request is coming from the JavaScript SDK.
+   <script>
+     function sendIMECommittedText(text) {
+       window.stream.sendIMECommittedText(text);
+     }
+     function sendIMEComposingText(text) {
+       window.stream.sendIMEComposingText(text);
+     }
+     function sendIMETextDeletion(counts) {
+       window.stream.sendIMETextDeletion(counts);
+     }
+     function sendIMEAction(name, params) {
+       window.stream.sendIMEAction(name, params);
+     }
+     function sendIMEComposingRegion(start, end) {
+       window.stream.sendIMEComposingRegion(start, end);
+     }
+   </script>
 
    ```
-   public class MainActivity extends AppCompatActivity implements AppInterface.ActionListener {
-   ...
 
-       /**
-        * Called as the request to open virtual keyboard
-        */
-       @Override
-       public void onOpenVirtualKeyboard() {
-       }
-
-       /**
-        * Called as the request to hide virtual keyboard
-        */
-       @Override
-       public void onHideVirtualKeyboard() {
-       }
+After you build your project, run the application and stream from Anbox Cloud,
+* when the text edit of the application in the container gains the focus, the client-side virtual keyboard pops up accordingly, and vice versa.
+* when typing with client-side virtual keyboard, the input text will be sent to the Android container and displayed in the running application.
 
-   ```
+The above workflow provides a general overview of how to integrate the client-side virtual keyboard feature into an Android application. For the complete implementation details, refer to the `enhanced_webview_streaming` example in the `example/android` folder.
 
-5. Set up the listener to the activity for which you want to capture the text input events from the virtual keyboard or monitor its visibility changes during streaming.
+Furthermore, by taking advantage of the interfaces that the AnboxWebView exposes, you can easily provide your own implementation of
```

Comment:
This should be in a separate section, something like `## Advanced` or `## Customizing the virtual keyboard`

Reply:
+1.
Will fix